### PR TITLE
build/pkgs/sagemath_{categories,environment,objects,repl}: Change to optional

### PR DESCRIPTION
**Pull request by mkoeppe**
... from "experimental". 

As suggested in https://github.com/sagemath/sage/issues/38190#issuecomment-2160546574

### :memo: Checklist

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Comment by kwankyu** [comment]
You need to edit `README.rst` files too.

**Comment by mkoeppe** [comment]
Good idea, done

File: pkgs/sagemath-objects/README.rst [hunk 1]
```diff
@@ -8,29 +8,26 @@ About SageMath
...
-distribution of a small part of the Sage Library.  Use at your own risk.  It
-provides a small, fundamental subset of the modules of the Sage library
+The pip-installable distribution package `sagemath-objects` is a
+distribution of a small part of the Sage Library.
+
+It provides a small, fundamental subset of the modules of the Sage library
 ("sagelib", `sagemath-standard`), making Sage objects, the element/parent
-framework, categories, the coercion system and the related metaclasses
-available.
+framework, categories, the coercion system and the related metaclasses.
```
Comment (kwankyu) [inline pkgs/sagemath-objects/README.rst]: `available.`

**Review by kwankyu — APPROVED** [review]
Thanks.

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**